File: Echo/LanguageSelectController.m
//
//  PHORLanguagesViewController.m
//  EnglishStudy
//
//  Created by Will Entriken on 6/3/12.
//  Copyright (c) 2012 __MyCompanyName__. All rights reserved.
//

#import "LanguageSelectController.h"
#import "Languages.h"
#import "GAI.h"
#import "GAIFields.h"
#import "GAIDictionaryBuilder.h"

@interface LanguageSelectController ()
@property (strong, nonatomic) NSArray *languages;
@end

@implementation LanguageSelectController
@synthesize languages = _languages;
@synthesize delegate = _delegate;

- (NSArray *)languages
{
    if (!_languages) {
        _languages = [Languages languages];
    }
    return _languages;
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    id tracker = [[GAI sharedInstance] defaultTracker];
    [tracker set:kGAIScreenName value:@"LanguageSelect"];
    [tracker send:[[GAIDictionaryBuilder createAppView] build]];

    if ([[[UIDevice currentDevice] systemVersion] compare:@"7.0" options:NSNumericSearch] != NSOrderedAscending)
        [self.tableView setContentInset:UIEdgeInsetsMake(20,
                                                         self.tableView.contentInset.left,
                                                         self.tableView.contentInset.bottom,
                                                         self.tableView.contentInset.right)];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.languages.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"leftDetail";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    cell.textLabel.text = (self.languages)[indexPath.row][@"tag"];
    cell.detailTextLabel.text = (self.languages)[indexPath.row][@"nativeName"];
    return cell;
}

#pragma mark - Table view delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [self.delegate languageSelectController:self didSelectLanguage:(self.languages)[indexPath.row][@"tag"] withNativeName:(self.languages)[indexPath.row][@"nativeName"]];
    [self dismissViewControllerAnimated:YES completion:nil];
}



@end

File: Echo/LanguageSelectController.swift
//
//  LanguageSelectController.swift
//  Echo
//
//  Created by William Entriken on 1/10/16.
//
//

import Foundation
import UIKit
import Google

protocol LanguageSelectControllerDelegate: class {
    func languageSelectController(controller: AnyObject, didSelectLanguage tag: String, withNativeName name: String) -> Void
}

class LanguageSelectController: UITableViewController {
    let languages = Languages.languages
    weak var delegate: LanguageSelectControllerDelegate?

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        let tracker = GAI.sharedInstance().defaultTracker
        tracker.set(kGAIScreenName, value: "LanguageSelect")
        tracker.send(GAIDictionaryBuilder.createScreenView().build() as [NSObject: AnyObject])
        self.tableView.contentInset = UIEdgeInsetsMake(20, self.tableView.contentInset.left, self.tableView.contentInset.bottom, self.tableView.contentInset.right)
    }
}

extension LanguageSelectController /*: UITableViewDataSource*/ {
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.languages.count
    }

    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let identifier = "leftDetail"
        let cell = tableView.dequeueReusableCellWithIdentifier(identifier)
        cell!.textLabel!.text = self.languages[indexPath.row].languageTag
        cell!.detailTextLabel!.text = self.languages[indexPath.row].nativeName
        return cell!
    }
}

extension LanguageSelectController /*: UITableViewDelegate*/ {
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        self.delegate?.languageSelectController(self, didSelectLanguage: self.languages[indexPath.row].languageTag, withNativeName: self.languages[indexPath.row].nativeName)
        self.dismissViewControllerAnimated(true, completion: { _ in })
    }
}
